### DISCUSSION TITLE
Cannot change workspace while using ImageMagick for taking screenshots.

### DISCUSSION THREAD

**Discussion by ogost** [i3/i3#4530] [Q&A] [answered]
I'm on Debian 11 with i3. Whenever I want to make a screenshot I use "import" from image magick, but I find myself unable to change workspace - keybindings don't work, and clicking on a workspace icon runs the actual screenshooting.
```
Debian 11
i3 version 4.19.1-non-git, 
import Version: ImageMagick 6.9.11-60 Q16

```
I'm not actually sure if it's related with ImageMagick or with i3. It used to work on Debian 10, maybe I should use something else for screenshooting?

**Comment by Airblader** [comment, marked as answer]
`import` waits for user input (see `man import`), and while it does, it maintains a grab, so bindings etc. won't work.

**Reply by ogost** [reply]
Indeed, it's a new behavior of image magick since v6.9. Thank you.

**Comment by graphitsign** [comment]
A workaround is to set sleep in front of import. I have write a script for that purpose, which copies the image to the clipboard as well:
 
    #!/bin/bash
    scr_folder="/tmp/screenshots"
    suffix="screenshot.png"
    datum=$(date '+%Y%m%d_%H%M%S')
    sleep 3 ; import ${scr_folder}/${datum}_${suffix}
    xclip -i ${scr_folder}/${datum}_${suffix} -t image/png -selection clipboard
